airflow/contrib/hooks/ssh_hook.py

```diff
@@ -14,125 +14,144 @@
+            if not self.username:
+                logging.info("username to ssh to host: {0} is not specified, using "
+                             "system's default provided by getpass.getuser()"
+                             .format(self.remote_host, self.ssh_conn_id))
+                self.username = getpass.getuser()
+
+            host_proxy = None
+            user_ssh_config_filename = os.path.expanduser('~/.ssh/config')
+            if os.path.isfile(user_ssh_config_filename):
+                ssh_conf = paramiko.SSHConfig()
+                ssh_conf.parse(open(user_ssh_config_filename))
+                host_info = ssh_conf.lookup(self.remote_host)
+                if host_info and host_info.get('proxycommand'):
+                    host_proxy = paramiko.ProxyCommand(host_info.get('proxycommand'))
+
+                if not (self.password or self.key_file):
+                    if host_info and host_info.get('identityfile'):
+                        self.key_file = host_info.get('identityfile')[0]
+
+            try:
+                client = paramiko.SSHClient()
+                client.load_system_host_keys()
+                if self.no_host_key_check:
+                    # Default is RejectPolicy
+                    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+                if self.password and self.password.strip():
+                    client.connect(hostname=self.remote_host,
+                                   username=self.username,
+                                   password=self.password,
+                                   timeout=self.timeout,
+                                   compress=self.compress,
+                                   sock=host_proxy)
+                else:
+                    client.connect(hostname=self.remote_host,
+                                   username=self.username,
+                                   key_filename=self.key_file,
+                                   timeout=self.timeout,
+                                   compress=self.compress,
+                                   sock=host_proxy)
+
+                self.client = client
+            except paramiko.AuthenticationException as auth_error:
+                logging.error("Auth failed while connecting to host: {0}, error: {1}"
+                              .format(self.remote_host, auth_error))
+            except paramiko.SSHException as ssh_error:
+                logging.error("Failed connecting to host: {0}, error: {1}"
+                              .format(self.remote_host, ssh_error))
+            except Exception as error:
+                logging.error("Error connecting to host: {0}, error: {1}"
+                              .format(self.remote_host, error))
+        return self.client
 
     @contextmanager
-    def tunnel(self, local_port, remote_port=None, remote_host="localhost"):
+    def create_tunnel(self, local_port, remote_port=None, remote_host="localhost"):
```

Comment:
no specific reason, i can keep it same.
actually SSHhook signature is changed with this new addition, and its not backward compatible so user will have to change the code anyway.